smalltalksrc/Slang/SLDeadCodeElimination.class.st

```diff
@@ -330,8 +324,8 @@ SLDeadCodeElimination >> mayMarkSendNode: aSendNode [
 	(aSendNode hasSideEffect not or: [
 		 aSendNode isConditionalSend or: [
 			 aSendNode isIterativeSend or: [
-				 (self isStructAccessorWithNoSideEffect: aSendNode) or: [
-					 aSendNode selector beginsWith: #cCoerce ] ] ] ]) ifTrue: [
+				 (aSendNode isStructAccessorWithNoSideEffectIn: codeGenerator)
+					 or: [ aSendNode selector beginsWith: #cCoerce ] ] ] ]) ifTrue: [
```

Comment:
Why moving this around? I don't know if we should put that behavior in the send node.

Reply:
Ok, eventually maybe we can move this to the code generator then